crates/electrum/src/bdk_electrum_client.rs

```diff
@@ -747,4 +749,59 @@ mod test {
 
         Ok(())
     }
+
+    #[cfg(feature = "default")]
+    #[test]
+    fn test_batch_fetch_anchors_reorg_uses_new_hash() -> anyhow::Result<()> {
+        let env = TestEnv::new()?;
+        let client = electrum_client::Client::new(env.electrsd.electrum_url.as_str()).unwrap();
+        let electrum_client = BdkElectrumClient::new(client);
+
+        env.mine_blocks(101, None)?;
+        env.wait_until_electrum_sees_block(Duration::from_secs(6))?;
+
+        let addr = env
+            .rpc_client()
+            .get_new_address(None, None)?
+            .assume_checked();
+        let txid = env.send(&addr, Amount::from_sat(50_000))?;
+        env.wait_until_electrum_sees_txid(txid, Duration::from_secs(6))?;
+
+        // Mine block that confirms transaction.
+        env.mine_blocks(1, None)?;
+        env.wait_until_electrum_sees_block(Duration::from_secs(6))?;
+        let height: u32 = env.rpc_client().get_block_count()? as u32;
+
+        // Populate header cache.
```

Comment:
Let's be more specific.


```suggestion
        // Add the pre-reorg block (that the tx is confirmed in) to the header cache.

```